src/modules/endpoint/models/CategoriesContainer.js

```diff
@@ -0,0 +1,72 @@
+import normalizeUrl from 'normalize-url'
+
+/**
+ * Contains a Map [string -> CategoryModel] and some helper functions
+ */
+class CategoriesContainer {
+  /**
+   * Creates a Map [url -> category] from the categories provided,
+   * whose parent attributes are first changed from id to url
+   * @param categories CategoryModels as array
+   */
+  constructor (categories = []) {
+    this._categories = new Map(categories.map(category => ([category.url, category])))
+  }
+
+  /**
+   * @return {CategoryModel[]} categories The categories as array
+   */
+  toArray () {
+    return Array.from(this._categories.values())
+  }
+
+  /**
+   * Returns the category with the given url
+   * @param url The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryByUrl (url) {
+    return this._categories.get(normalizeUrl(url))
+  }
+
+  /**
+   * Returns the category with the given id
+   * @param id The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryById (id) {
+    return this.toArray().find(category => category.id === Number(id))
+  }
+
+  /**
+   * Returns all children of the given category
+   * @param category The category
+   * @return {CategoryModel[] | undefined} The children
+   */
+  getChildren (category) {
+    if (category) {
```

Comment:
better throw an error here is category is undefined. Or just ignore it and let the language handle it.
You could also return an empty array but that it also somewhat not intuitive :)

Reply:
we only use getChildren in CategoriesPage getContent(), where category can't be undefined, so category should always be defined

Reply:
so I would just remove the if (category), and the same for getAncestors

Reply:
@maxammann okay?

Reply:
getChildren() is also called in PdfFetcherPage, but category === undefined is also not possible here